cpp/inc/bond/core/exception.h

```diff
@@ -55,11 +57,43 @@ BOND_NORETURN inline void MergerContainerException(uint32_t payload, uint32_t ob
 }
 
 
+BOND_NORETURN inline void InvalidKeyTypeException(void)
+{
+    BOND_THROW(CoreException,
+        "Map key type not valid");
+}
+
+namespace detail
+{
+    template <typename Key>
+    BOND_NORETURN inline void ElementNotFoundExceptionHelper(const Key& key, typename boost::enable_if<is_wstring<Key>>::type* = nullptr)
+    {
+        try
+        {
+            BOND_THROW(
+                CoreException,
+                "Map element not found: key: " <<
+                    boost::locale::conv::utf_to_utf<char>(key.c_str(), key.c_str() + key.size(),  boost::locale::conv::stop));
```

Comment:
We need to access the string via the [string concept](https://microsoft.github.io/bond/manual/bond_cpp.html#string-concept) here.

    boost::locale::conv::utf_to_utf(
        string_data(key),
        string_data(key) + string_length(key),
        boost::locale::conv::stop);